src/encoder.c

```diff
@@ -49,39 +53,55 @@ int32_t get_encoder_right_diff_count(void)
 }
 
 /**
+ * @brief Read left motor encoder total count.
+ *
+ * The total count is simply the sum of all encoder counter differences.
+ */
+int32_t get_encoder_left_total_count(void)
+{
+	return left_total_count;
+}
+
+/**
+ * @brief Read right motor encoder total count.
+ *
+ * The total count is simply the sum of all encoder counter differences.
+ */
+int32_t get_encoder_right_total_count(void)
+{
+	return right_total_count;
+}
+
+/**
  * @brief Read left motor encoder travelled distance in millimeters.
  */
-float get_encoder_left_distance(void)
+int32_t get_encoder_left_micrometers(void)
 {
-	return left_distance;
+	return left_micrometers;
 }
 
 /**
  * @brief Read right motor encoder travelled distance in millimeters.
  */
-float get_encoder_right_distance(void)
+int32_t get_encoder_right_micrometers(void)
 {
-	return right_distance;
+	return right_micrometers;
 }
 
 /**
- * @brief Read left motor encoder total count.
- *
- * The total count is simply the sum of all encoder counter differences.
+ * @brief Read left motor speed in meters per second.
  */
-uint16_t get_encoder_left_total_count(void)
+float get_encoder_left_speed(void)
 {
-	return left_total_count;
+	return left_speed;
 }
 
 /**
- * @brief Read right motor encoder total count.
- *
- * The total count is simply the sum of all encoder counter differences.
+ * @brief Read right motor speed in meters per second.
  */
-uint16_t get_encoder_right_total_count(void)
+float get_encoder_right_speed(void)
 {
-	return right_total_count;
+	return right_speed;
 }
 
 /**
```

Comment:
Update the header with distance and speed calculations and delete the comments inside the function.